lib/optimove.dart

```diff
@@ -0,0 +1,255 @@
+import 'dart:async';
+import 'package:flutter/services.dart';
+import 'dart:io' show Platform;
+
+class Optimove {
+  static const MethodChannel _methodChannel = MethodChannel('optimove_flutter_sdk');
+  static const EventChannel _eventChannel = EventChannel('optimove_flutter_sdk_events');
+  static StreamSubscription? _eventStream;
+
+  static void Function(OptimovePushNotification)? _pushOpenedHandler;
+  static void Function(OptimovePushNotification)? _pushReceivedHandler;
+  static void Function(OptimoveDeepLinkOutcome)? _deepLinkHandler;
+  static Function? _inboxUpdatedHandler;
+  static void Function(Map<String, dynamic>)? _inAppDeepLinkHandler;
+
+  static Future<void> registerUser({required String userId, required String email}) async {
+    return _methodChannel.invokeMethod('registerUser', {'userId': userId, 'email': email});
+  }
+
+  static Future<void> setUserId({required String userId}) async {
+    return _methodChannel.invokeMethod('setUserId', {'userId': userId});
+  }
+
+  static Future<void> setUserEmail({required String email}) async {
+    return _methodChannel.invokeMethod('setUserEmail', {'email': email});
+  }
+
+  static Future<String> getVisitorId() async {
+    String visitorId = await _methodChannel.invokeMethod('getVisitorId');
+    return visitorId;
+  }
+
+  static Future<void> reportEvent({required String event, Map<String, dynamic>? parameters}) async {
+    return _methodChannel.invokeMethod('reportEvent', {'event': event, 'parameters': parameters});
+  }
+
+  static Future<void> reportScreenVisit({required String screenName, String? screenCategory}) async {
+    return _methodChannel.invokeMethod('reportScreenVisit', {'screenName': screenName, 'screenCategory': screenCategory});
+  }
+
+  static Future<String?> getUserId() async {
+    String? userId = await _methodChannel.invokeMethod('getUserId');
+    return userId;
+  }
+
+  static void pushRequestDeviceToken() {
+    if (Platform.isIOS) {
+      _methodChannel.invokeMethod('pushRequestDeviceToken');
+    } else if (Platform.isAndroid) {
+      _methodChannel.invokeMethod('pushRegister');
+    }
+  }
+
+  static void enableStagingRemoteLogs() {
+    if (Platform.isAndroid) {
+      _methodChannel.invokeMethod('enableStagingRemoteLogs');
+    }
```

Comment:
Unimplemented for iOS?

Reply:
There is no method for this in iOS, not sure if will ever be. Can change the method name to reflect the Android only implementation